### PR TITLE
Sort audiences alphabetically

### PR DESCRIPTION
**Pull request by voneiden**
Sort order for audiences was previously hardcoded. This commit changes it to locale based alphabetical sorting.

Ref LINK-1461

### CODE REVIEW

File: src/domain/event/hooks/useAudienceOptions.ts [hunk 1]
```diff
@@ -6,15 +8,15 @@ import { sortAudienceOptions } from './utils';
...
-      .sort(sortAudienceOptions) || [];
-
-  return audienceOptions;
+  return useMemo(() => {
+    return (
+      audienceData?.keywordSet?.keywords
+        ?.map((keyword) => getKeywordOption({ keyword, locale }))
+        .sort(sortAudienceOptions) || []
+    );
+  }, [audienceData, locale]);
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/hooks/useAudienceOptions.ts:19]: You can also use `lodash/sortBy` function to sort audiences.
```
return sortBy(
      audienceData?.keywordSet?.keywords
        ?.map((keyword) => getKeywordOption({ keyword, locale }))
        .sort(sortAudienceOptions) || [],
      ['label']
    );
```

File: src/domain/event/hooks/utils.ts [hunk 2]
```diff
@@ -38,13 +36,8 @@ export const getEventUpdateAction = (
...
 };
 
-const getAudienceIndex = (atId: string) => {
-  const index = AUDIENCE_ORDER.indexOf(getValue(parseIdFromAtId(atId), ''));
-  return index !== -1 ? index : AUDIENCE_ORDER.length;
-};
-
 export const sortAudienceOptions = (a: OptionType, b: OptionType): number =>
-  getAudienceIndex(a.value) - getAudienceIndex(b.value);
+  a.label.localeCompare(b.label);
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/hooks/utils.ts:40]: Or one step further you could have
```
export const getSortAudienceOptions =
  (locale: Language) =>
  (a: OptionType, b: OptionType): number =>
    a.label.localeCompare(b.label, locale);
```
function and use it like this:
```
audienceData?.keywordSet?.keywords
      ?.map((keyword) => getKeywordOption({ keyword, locale }))
      .sort(getSortAudienceOptions(locale)) || []
```

**Comment by jorilindell** [comment]
**should show 10 first audiences by default and rest by clicking show more** tests seems to be failing. That test should be tested against sorted audience data.

File: src/domain/event/hooks/useAudienceOptions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/hooks/useAudienceOptions.ts:19]: You are right. Audience data should be same after fetching it from API. Topics keyword set depends on selected event type. But that's a different story.

[the remark this answers is not in the document]
Comment (voneiden) [reply src/domain/event/hooks/useAudienceOptions.ts:19]: A little bit of profiling on Firefox (YMMV):  without useMemo useAudienceOptions takes on average 5 ms on every render, with useMemo it takes 1 ms. 

The form is already rather heavy, a single keypress takes 350 - 450 ms to process on my setup. OK, 4 ms makes that like 1% slower which is marginal in the big picture..

[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/hooks/useAudienceOptions.ts:19]: Ok, better to have useMemo in that case.

**Review by jorilindell — APPROVED** [review]
Looks good and performance improvements are always welcome. Well done!

LGTM

**Pull request merged**